src/story/prologue.cpp

```diff
@@ -20,18 +25,38 @@ void Prologue()
 		"boss_battle");
 	utils::HandleSound(utils::SoundOperations::Play, "boss_battle repeat");
 
-	MainCharacter main_character("Traveler");
-	main_character.PrintStats();
+	if (file)
+	{
+		LoadSavedGame(main_character);
```

Comment:
Only the 'LoadSavedGame' line should be inside the if. Everything below it is exactly the same as in the 'else' block, hence we can get rid of it and have one instance of the battle sequence in the code.